docs/adr/0009-publish-chart-to-rancher-charts.md

```diff
@@ -0,0 +1,95 @@
+
+There are two annotations: `catalog.cattle.io/rancher-version` and `catalog.cattle.io/kube-version` which must include lower and upper bounds. Since this may vary from one Rancher major version to another, the easiest approach would be to have this removed from the original chart and manually published to `rancher/charts` when a new Rancher major version is released (refer to [fleet chart](https://github.com/rancher/fleet/blob/master/charts/fleet/Chart.yaml)). Subsequent updates to the chart will just use the existing bounds.
+
+#### Chart dependencies
+
+Dependencies must be added to the Rancher file `pkg/image/origins.go` which denotes the source code repository of each image used in chart.
+
+Rancher Turtles has Cluster API Operator as a dependency, which also adds the Cert Manager dependency. `cert-manager` defines a `kubeVersion` field in `Chart.yaml` which would force us to annotate the chart (refer to [here](https://github.com/rancher/charts/blob/dev-v2.8/charts/rancher-monitoring/103.0.4%2Bup45.31.1/charts/grafana/Chart.yaml)). This is probably a workaround we would like to avoid.
+
+### Option 2 - specific rancher-turtles repository.
+
+So far, we've used the `rancher-turtles` repository as a Helm repository using [helm/chart-releaser](https://github.com/helm/chart-releaser). This is serving the chart from the URL https://rancher-sandbox.github.io/rancher-turtles. We could continue using this mechanism for hosting the chart while using a more production-ready CNAME for the repository:  https://charts.rancher-turtles.com/. In preparation for Turtles to be generally available, we must avoid using the `rancher-sandbox` name.
+
+This means that the release workflow manages the end-to-end release process, including publishing the Helm chart, which will automatically become available after the release is created. On the other hand, it separates the chart from Rancher even though Rancher Turtles can only be installed together with Rancher. This differs from other projects that use a similar approach for chart management, like Kubewarden that is technically independent of Rancher.
+
+## Consequences
+
+### Using `rancher-charts`
+- The existing publish mechanism based on `chart-releaser` will be deprecated.
+- Hard to achieve a high level of automation.
+- Chart repository is accessible from Rancher Marketplace.
+- Release process will now require an extra step to create the `rancher/charts` PR: documentation must be aligned with the new configuration.
+- First chart release for each major version of Rancher will require a manual update to the charts to set the correct version bounds as annotations. Subsequent updates should be applied via GitHub Actions.
+    - Fixing dependency charts.
+- This change should be transparent to the user.
+
+### Using specific repository
+- The existing publish mechanism will remain.
+- Already fully automated
+- Chart repository must be added before installing from Rancher Marketplace.
```

Comment:
In the future we can follow the kubewarden approach and get the UI extension to add the repository.

Reply:
I updated the description with a mention to using the Kubewarden approach in the future.